cmd/influx/v1_dbrp.go

```diff
@@ -0,0 +1,386 @@
+package main
+
+import (
+	"context"
+	"io"
+
+	"github.com/influxdata/influxdb/v2"
+	"github.com/influxdata/influxdb/v2/cmd/influx/internal"
+	"github.com/influxdata/influxdb/v2/dbrp"
+	"github.com/spf13/cobra"
+)
+
+func cmdV1DBRP(f *globalFlags, opt genericCLIOpts) *cobra.Command {
+	cmd := opt.newCmd("dbrp", nil, false)
+	cmd.Aliases = []string{"dbrp"}
+	cmd.Short = "Database retention policy mappings for v1 APIs"
+	cmd.Run = seeHelp
+
+	cmd.AddCommand(
+		v1DBRPCreateCmd(f, opt),
+		v1DBRPFindCmd(f, opt),
+		v1DBRPDeleteCmd(f, opt),
+		v1DBRPUpdateCmd(f, opt),
+	)
+
+	return cmd
+}
+
+var v1DBRPCRUDFlags struct {
+	id          string
+	json        bool
+	hideHeaders bool
+}
+
+var v1DBRPFindFlags struct {
+	BucketID string       // Specifies the bucket ID to filter on
+	DB       string       // Specifies the database to filter on
+	Default  bool         // Specifies filtering on default
+	ID       string       // Specifies the mapping ID to filter on
+	Org      organization // required  // Specifies the organization ID to filter on
+	RP       string       // Specifies the retention policy to filter on
+}
+
+func v1DBRPFindCmd(f *globalFlags, opt genericCLIOpts) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "list",
+		Short: "List DBRP",
+		RunE:  checkSetupRunEMiddleware(&flags)(v1DBRPFindF),
+	}
+
+	f.registerFlags(opt.viper, cmd)
+	registerPrintOptions(opt.viper, cmd, &v1DBRPCRUDFlags.hideHeaders, &v1DBRPCRUDFlags.json)
+	v1DBRPFindFlags.Org.register(opt.viper, cmd, false)
+	cmd.Flags().StringVarP(&v1DBRPCreateFlags.BucketID, "bucket-id", "b", "", "the bucket ID to filter on")
+	cmd.Flags().StringVarP(&v1DBRPCreateFlags.DB, "db", "d", "", "the v1 database to map from")
+	cmd.Flags().BoolVar(&v1DBRPCreateFlags.Default, "default", false, "Specify if this mapping represents the default retention policy for the database specificed")
```

Comment:
Does `--default` == the magic "autogen" RP? If so, should we validate up-front that `--rp` isn't passed when `--default` is used?

In the V1 write-path, if no RP is specified in a request then "autogen" is used. Could we mimic that behavior here (instead of having a `--default` flag) for symmetry?

Reply:
As far as I can tell Default is a property of the mapping not the RP.